src/init.cpp

```diff
@@ -925,7 +925,8 @@ bool AppInitParameterInteraction()
         // Special-case: if -debug=0/-nodebug is set, turn off debugging messages
         const std::vector<std::string> categories = gArgs.GetArgs("-debug");
 
-        if (find(categories.begin(), categories.end(), std::string("0")) == categories.end()) {
+        if (std::none_of(categories.begin(), categories.end(),
+            [](std::string cat){return cat == std::string("0") || cat == std::string("none");})) {
```

Comment:
```cpp
[](const auto& cat){return cat == "0" || cat == "none";}
```

Reply:
@promag thank you for your review.
1.  auto specifier can not use in lambda epression when -std=c++11.
2. std::string has == operater with c-string.

so I will change to following:
``` [](std::string cat){return cat == "0" || cat == "none";})) { ```

Reply:
BTW, this is not the 1st time I would prefer to read:
```cpp
std::set<std::string> categories = gArgs.GetUniqueArgs("-debug");

if (!categories.count("0") && !categories.count("none")) {
```
Or simply:
```cpp
if (!gArgs.IsArgSet("-debug", "0") && !gArgs.IsArgSet("-debug", "none")) {
```